### PR TITLE
Initialize constant postprocessor value in constructor

### LINKED ISSUE #27567: ConstantPostprocessor value can/should be populated from the point of construction
### Motivation

@Naktakala experienced a lot of confusion when attempting to retrieve a value corresponding to a `ConstantPostprocessor` and finding it to be 0

### Design

Populate the reporter value in the constructor

### Impact

Eliminate unnecessary user confusion

### PR DESCRIPTION
**Pull request by lindsayad**
Closes #27567

### CODE REVIEW

**Comment by lindsayad** [comment]
Added failing test with link to civet to https://github.com/idaholab/moose/issues/27555

**Comment by lindsayad** [comment]
@YaqiWang this change appears to affect a Griffin test that uses `ConstantPostprocessor`. Could you take a look?

File: framework/src/postprocessors/ConstantPostprocessor.C [hunk 1]
```diff
@@ -25,6 +25,12 @@ ConstantPostprocessor::validParams()
 ConstantPostprocessor::ConstantPostprocessor(const InputParameters & params)
   : GeneralPostprocessor(params), _value(getParam<Real>("value"))
 {
+  const PostprocessorReporterName r_name(name());
+  auto & write_data = _fe_problem.getReporterData(ReporterData::WriteKey());
+
+  // Initialize values
+  const auto value = getParam<Real>("value");
+  write_data.setReporterValue<PostprocessorValue>(r_name, value);
```
Comment (GiudGiud) [inline framework/src/postprocessors/ConstantPostprocessor.C:33]:
```suggestion
  write_data.setReporterValue<PostprocessorValue>(r_name, _value);
```

Comment (GiudGiud) [reply framework/src/postprocessors/ConstantPostprocessor.C:33]: if we want the control to be able to mess with it on initial ?

Comment (lindsayad) [reply framework/src/postprocessors/ConstantPostprocessor.C:33]: `getParam<Real>("value")` represents the same thing as `_value`, a reference to the controllable parameter in the input warehouse. But might as well do the change for cleaner DRY code

**Review by GiudGiud — APPROVED** [review]
pending griffin fix

**Comment by YaqiWang** [comment]
> @YaqiWang this change appears to affect a Griffin test that uses `ConstantPostprocessor`. Could you take a look?

Sorry, I did not notice this. It is affecting a test added by @jthano. It might reveal a tiny bug on initialization.

**Comment by jthano** [comment]
Just merged the fix. Sorry about that, this is a good change for sure
https://github.inl.gov/ncrc/griffin/pull/1921

**Pull request merged**